src/node_platform.cc

```diff
@@ -332,11 +332,17 @@ int NodePlatform::NumberOfWorkerThreads() {
 
 void PerIsolatePlatformData::RunForegroundTask(std::unique_ptr<Task> task) {
   Isolate* isolate = Isolate::GetCurrent();
-  HandleScope scope(isolate);
+#ifdef DEBUG
+  SealHandleScope scope(isolate);
+#endif
   Environment* env = Environment::GetCurrent(isolate);
-  InternalCallbackScope cb_scope(env, Local<Object>(), { 0, 0 },
-                                 InternalCallbackScope::kAllowEmptyResource);
-  task->Run();
+  if (env != nullptr) {
+    InternalCallbackScope cb_scope(env, Local<Object>(), { 0, 0 },
+                                   InternalCallbackScope::kAllowEmptyResource);
+    task->Run();
+  } else {
+    task->Run();
```

Comment:
Can `task->Run()` be moved out from if statement ? Then we just write this line once :)

Reply:
@ZYSzys Yeah, it’s a bit unfortunate, but moving it out would mean that it is never executed inside the callback scope, not even if we want it to be.

Reply:
Oh, I see. Thanks for your explanation.